scripts/gas-refund-program/transactions-indexing/successful-swap-tx-indexing.ts

```diff
@@ -46,17 +45,16 @@ export async function computeSuccessfulSwapsTxFeesRefund({
     `swapTracker start indexing between ${startTimestamp} and ${endTimestamp}`,
   );
 
-  const [accPendingGasRefundByAddress, veryLastTimestampProcessed] =
-    await Promise.all([
-      fetchPendingGasRefundData({ chainId, epoch }),
-      fetchVeryLastTimestampProcessed({ chainId, epoch }),
-    ]);
+  const veryLastTimestampProcessed = await fetchVeryLastTimestampProcessed({ chainId, epoch })
 
   const _startTimestamp = Math.max(
     startTimestamp,
     veryLastTimestampProcessed + 1,
   );
 
+  // load all past swaps into memory as a record to track and de-duplicate txs
+  const pastTXs: Record<string, number> = await fetchTransactionOccurences(epoch, chainId)
```

Comment:
I think since this change https://github.com/paraswap/paraswap-volume-tracker/commit/921f52f87d41300f39401a1504029b3f5ebf7e7c, duplicates should never after epoch 12.
Please verify, if so that might alleviate some of logic.

Reply:
in theory yes, but depends on whether we want to trust external data. I think we should be as careful as possible.

Reply:
just checked, in the previous case of tx 0xb10c51756678cb6cb1635ac339f9caa592f49ea6da936b109dbd49da8e0e0a6a those txs don't look like they'd have made it through, not the first or it's subsequent tx contained swaps.

![image](https://user-images.githubusercontent.com/6588340/166471142-cb924d3a-7ea0-4065-bf0a-a9def8da0da4.png)

Reply:
Yep it's the intention of that cond